qiskit/transpiler/passes/crosstalk_adaptive_schedule.py

```diff
@@ -0,0 +1,667 @@
+        """
+        backend_prop = self.backend_prop
+        qid = 0
+        for qinfo in backend_prop.qubits:
+            for item in qinfo:
+                if item.name == "T1":
+                    # Convert us to ns
+                    self.bp_t1_time[qid] = int(item.value*1000)
+                elif item.name == "T2":
+                    self.bp_t2_time[qid] = int(item.value*1000)
+            qid += 1
+        for ginfo in backend_prop.gates:
+            if ginfo.gate == 'u1':
+                q_0 = ginfo.qubits[0]
+                for item in ginfo.parameters:
+                    if item.name == 'gate_error':
+                        if item.value == 1.0:
+                            self.bp_u1_err[q_0] = 0.999999
+                        else:
+                            self.bp_u1_err[q_0] = round(item.value, NUM_PREC)
+                    elif item.name == "gate_length":
+                        self.bp_u1_dur[q_0] = int(item.value)
+            elif ginfo.gate == 'u2':
+                q_0 = ginfo.qubits[0]
+                for item in ginfo.parameters:
+                    if item.name == 'gate_error':
+                        if item.value == 1.0:
+                            self.bp_u2_err[q_0] = 0.999999
+                        else:
+                            self.bp_u2_err[q_0] = round(item.value, NUM_PREC)
+                    elif item.name == "gate_length":
+                        self.bp_u2_dur[q_0] = int(item.value)
+            elif ginfo.gate == 'u3':
+                q_0 = ginfo.qubits[0]
+                for item in ginfo.parameters:
+                    if item.name == 'gate_error':
+                        if item.value == 1.0:
+                            self.bp_u3_err[q_0] = 0.999999
+                        else:
+                            self.bp_u3_err[q_0] = round(item.value, NUM_PREC)
+                    elif item.name == "gate_length":
+                        self.bp_u3_dur[q_0] = int(item.value)
+            elif ginfo.gate == 'cx':
+                q_0 = ginfo.qubits[0]
+                q_1 = ginfo.qubits[1]
+                r_0 = min(q_0, q_1)
+                r_1 = max(q_0, q_1)
+                for item in ginfo.parameters:
+                    if item.name == 'gate_error':
+                        if item.value == 1.0:
+                            self.bp_cx_err[(r_0, r_1)] = 0.999999
+                        else:
+                            self.bp_cx_err[(r_0, r_1)] = round(item.value, NUM_PREC)
+                    elif item.name == "gate_length":
+                        self.bp_cx_dur[(r_0, r_1)] = int(item.value)
+
+    def cx_tuple(self, gate):
+        """
+        Representation for two-qubit gate
+        """
+        physical_q_0 = gate.qargs[0].index
+        physical_q_1 = gate.qargs[1].index
+        r_0 = min(physical_q_0, physical_q_1)
+        r_1 = max(physical_q_0, physical_q_1)
+        return (r_0, r_1)
```

Comment:
Yes I assume that because the 20-qubit devices report the same error in both directions. If error rates are only slightly different in both directions that shouldn't affect the overall logic. If crosstalk behavior is different we may to have make changes.

Reply:
Added a comment.

Reply:
ok